### PR TITLE
Added a sample for RTDB transaction support

### PR DESCRIPTION
**Pull request by hiranya911**
Sample to appear at https://firebase.google.com/docs/database/admin/save-data#section-transactions

### CODE REVIEW

File: db/index.py [hunk 1]
```diff
@@ -189,6 +189,15 @@ def get_push_key():
     # [END push_key]
     print(post_id)
 
+def run_transaction():
+    # [START transaction]
+    def increment_votes(current_value):
+        return current_value + 1 if current_value else 1
+
+    upvotes_ref = db.reference('server/saving-data/fireblog/posts/-JRHTHaIs-jNPLXOQivY/upvotes')
+    upvotes_ref.transaction(increment_votes)
```
Comment (abehaskins) [inline db/index.py]: Is there a failure condition here? How do you catch that?

Comment (hiranya911) [inline db/index.py]: `transaction()` returns a boolean. Should we demonstrate that in the sample?

**Comment by hiranya911** [comment]
This API underwent a minor change. Updated the sample accordingly, and also showing how to handle return value and exceptions.

File: db/index.py [hunk 2]
```diff
@@ -189,6 +189,19 @@ def get_push_key():
     # [END push_key]
     print(post_id)
 
+def run_transaction():
+    # [START transaction]
+    def increment_votes(current_value):
+        return current_value + 1 if current_value else 1
+
+    upvotes_ref = db.reference('server/saving-data/fireblog/posts/-JRHTHaIs-jNPLXOQivY/upvotes')
```
Comment (samtstern) [inline db/index.py]: I'd probably make run_transaction take `upvotes_ref` as a parameter rather than having this strange hard-coded path

Comment (hiranya911) [inline db/index.py]: `run_transaction()` is a method defined on the `db.Reference` object. This is similar to the corresponding Node.js and Java APIs: https://firebase.google.com/docs/database/admin/save-data#section-transactions

Note that the path comes from the existing code samples.

**Pull request merged**